modules/gui/src/com/haulmont/cuba/gui/export/ExcelExporter.java

```diff
@@ -526,7 +536,13 @@ protected int createGroupRow(GroupTable table, List<Table.Column> columns, int r
                 createRow(table, columns, groupNumber, ++rowNumber, itemId);
             }
         }
-        sheet.groupRow(oldRowNumber + 1, rowNumber);
+
+        if (checkIsRowNumberExceed(rowNumber)) {
+            sheet.groupRow(oldRowNumber + 1, MAX_ROW_COUNT);
```

Comment:
If row number exceeded why we add group row? What is going on here?

Reply:
`sheet.groupRow` means create a group row from the `oldRowNumber+1` position to `rowNumber` position. So, if `rowNumber` is exceeded we should cut it by `MAX_ROW_COUNT`.